cmd/registry/cmd/apply/apply.go

```diff
@@ -47,26 +47,31 @@ func Command() *cobra.Command {
 			}
 
 			ctx := cmd.Context()
+			c, err := connection.ActiveConfig()
+			if err != nil {
+				return err
+			}
 			if project == "" {
-				c, err := connection.ActiveConfig()
-				if err != nil {
-					return err
-				}
 				project, err = c.ProjectWithLocation()
 				if err != nil {
 					return fmt.Errorf("%s: please use --parent or set registry.project in configuration", err)
 				}
 			} else if !strings.Contains(project, "/locations/") {
 				project += "/locations/global"
 			}
-			client, err := connection.NewRegistryClient(ctx)
+
+			client, err := connection.NewRegistryClientWithSettings(ctx, c)
```

Comment:
nit: I was a little confused by `WithSettings` and looked up the function to see that `Settings` here is synonymous with `Config`. Could we change that to `WithConfig` in some future PR? I don't think it's important enough to delay this or the next PR that depends on this one, so no hurry.